internal/transport/http2_client.go

```diff
@@ -1023,6 +1023,12 @@ func (t *http2Client) Close(err error) {
 	case <-timer.C:
 		t.logger.Infof("Failed to write a GOAWAY frame as part of connection close after %s. Giving up and closing the transport.", goAwayLoopyWriterTimeout)
 	}
+	select {
```

Comment:
we can just have single comment above select. Something like `// wait for the reader channel to close before proceeding further`

In the default case, we should probably log if running in a debug mode. Something similar to what we do in case of writerDone

if t.logger.V(logLevel) {
		t.logger.Infof("Closing: %v", err)
	}